parsers/cxx/cxx_parser_variable.c

```diff
@@ -502,8 +508,31 @@ boolean cxxParserExtractVariableDeclarations(
 			// remove the identifier
 			cxxTokenChainTakeRecursive(pChain,pIdentifier);
 
-			// FIXME: We should check the contents of square parentheses and
-			//        empty them if they contains something that is not a numeric constant.
+			// Fix square parentheses: if they contain something that is not a numeric
+			// constant then empty them up
+			CXXToken * pPartOfType = t->pPrev;
+			CXX_DEBUG_ASSERT(pPartOfType,"There should be a part of type name here");
+
+			while(pPartOfType && cxxTokenTypeIs(pPartOfType,CXXTokenTypeSquareParenthesisChain))
+			{
+				CXXTokenChain * pAuxChain = pPartOfType->pChain;
+			
```

Comment:
One of the reason I asked is that git diff renders the whitespaces with RED.
It is noise for reviewing task. I may turn it off. However, someone, a potential contributor may make a pull request what trims the whitespaces. We, you, I, and the contributor should not take time for such patch. Only the programmer introducing new code can avoid the patch.

If you were using emacs, `M-x whitespace-cleanup` helps you. However, you have your own editor/IDE.